src/utils/checkPermissions.js

```diff
@@ -0,0 +1,11 @@
+const CustomError = require("../errors");
+
+const checkPermissions = (requestUser, resourceUserId) => {
+  if (requestUser.role === "admin") return;
+  if (requestUser.userId === resourceUserId.toString()) return;
+  throw new CustomError.UnauthorizedError(
+    "Not authorized to access this route"
+  );
+};
+
+module.exports = checkPermissions;
```

Comment:
Good job with all the attention being paid to user authentication, very important